tests/unit/test_repos_branch_protection.py

```diff
@@ -0,0 +1,207 @@
+"""Unit tests for methods implemented on Branch Protection."""
+import github3
+from github3.repos.branch import BranchProtection
+from . import helper
+from json import dumps
+import pytest
+
+protection_example_data = helper.create_example_data_helper(
+    'branch_protection_example'
+)
+protection_url_for = helper.create_url_helper(
+    'https://api.github.com/repos/octocat/Hello-World/'
+    'branches/master/protection'
+)
+
+
+# class TestBranchProtection(helper.UnitRequiresAuthenticationHelper):
```

Comment:
Why is this commented out?

Reply:
Basically the BranchProtection endpoint is duplicating the functionality of the separate endpoints that we have in separate classes. Would you suggest "double-coverage" of that functionality?
Seems like github will eventually deprecate and remove one of the ways of setting branch protection either via single endpoint or via multiple endpoints.